core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXV1RequestToStackV4RequestConverter.java

```diff
@@ -168,7 +168,7 @@ private void validateSubnetIds(NetworkV4Request network, DetailedEnvironmentResp
     }
 
     private void validateSubnet(NetworkV4Request network, DetailedEnvironmentResponse environment, String subnetId) {
-        if (!environment.getNetwork().getSubnetIds().contains(subnetId)) {
+        if (subnetId != null && !environment.getNetwork().getSubnetIds().contains(subnetId)) {
```

Comment:
Minor one: I know that this is old code, but since there are 3 dots in environment.getNetwork().getSubnetIds().contains(...), this means that we can get NPE from 3 places. I prefer a bit more defensive programming approach